sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -962,9 +962,9 @@ def upload_blob(
         ) -> BlobClient:
         """Creates a new blob from a data source with automatic chunking.
 
-        :param name: The blob with which to interact. If specified, this value will override
+        :param str name: 
+            The blob with which to interact. If specified, this value will override
             a blob value specified in the blob URL.
```

Comment:
Technically outside your changes but: Not sure which "blob URL" this comment is referring to? Maybe this is a copied comment from BlobClient and we should remove that sentence about overriding?

This would apply to a lot of places in this review.

Reply:
Yeah, I found this to be odd as well. Can remove it on my next pass-through after we receive some more feedback 😄